tests/http/federation/test_matrix_federation_agent.py

```diff
@@ -466,7 +466,10 @@ def _do_get_via_proxy(
         else:
             assert isinstance(proxy_server_transport, FakeTransport)
             client_protocol = proxy_server_transport.other
+            assert isinstance(client_protocol, Protocol)
             c2s_transport = client_protocol.transport
+            assert c2s_transport is not None
+            assert isinstance(c2s_transport, FakeTransport)
```

Comment:
It seems like a regression in mypy(-zope?) that I need to first say it is an `ITransport` (with the `not None` assert), then assert the concrete type.